src/libaktualizr/uptane/uptanerepository.cc

```diff
@@ -47,6 +47,45 @@ bool RepositoryCommon::verifyRoot(const std::string& root_raw) {
 
 void RepositoryCommon::resetRoot() { root = Root(Root::Policy::kAcceptAll); }
 
+bool RepositoryCommon::updateRoot(INvStorage& storage, const IMetadataFetcher& fetcher,
```

Comment:
Cool, that exactly was I thought about when I saw the initial implementation - share as much code as possible between the director and imagerepo implementation.

Perhaps, it makes sense to add a reference(s) to the corresponding paragraphs of the uptane spec.